### PR TITLE
Make all nl_func_call_* options be mutually compatible

### PR DESCRIPTION
**Pull request by fwojcik**
Before this, `nl_func_call_args` would prevent `nl_func_call_start` from working

Also, if `nl_func_call_args` was ignore, then `nl_func_call_start` would prevent
the following options from working:
- `nl_func_call_paren`
- `nl_func_call_paren_empty`
- `nl_func_call_empty`
- `nl_func_call_start_multi_line`
- `nl_func_call_args_multi_line`
- `nl_func_call_end_multi_line`

This also incidentally fixes a weird edge case where `nl_func_call_args` would not work for user function calls if neither `nl_func_call_start` nor any other option in that list were enabled.

Without this fix, the included unit tests fail as follows:
```diff
12/14 Test  #2: c ................................***Failed   63.61 sec
Tests: ['c']
Processing /home/fwojcik/progs/uncrustify/tests/c.test
diff --git a/home/fwojcik/progs/uncrustify/tests/expected/c/00730-nl_func_call.c b/home/fwojcik/progs/uncrustify/build/tests/results/c/00730-nl_func_call.c
index 38c17e45..02e9b49d 100644
--- a/home/fwojcik/progs/uncrustify/tests/expected/c/00730-nl_func_call.c
+++ b/home/fwojcik/progs/uncrustify/build/tests/results/c/00730-nl_func_call.c
@@ -1,21 +1,23 @@
 void example(int a, int b, int c, int d);
 void example2(void);
-void userfunc
-(
-       int w,
-       int x,
-       int y,
-       int z
-);
+void userfunc(
+       int w, int x, int y, int z
+       );
 
 int main(void) {
        example
-       (
-               1,
-               2,
-               3,
-               4
-       );
+               (1,
+               2,
+               3,
+               4
+               );
+
+       example
+               (1,
+               2,
+               3,
+               4
+               );
 
        example
        (
@@ -34,20 +36,11 @@ int main(void) {
        );
 
        example
-       (
-               1,
-               2,
-               3,
-               4
-       );
-
-       example
-       (
-               1,
-               2,
-               3,
-               4
-       );
+               (1,
+               2,
+               3,
+               4
+               );
 
        example
        (
@@ -81,59 +74,43 @@ int main(void) {
        (
        );
 
-       userfunc
-       (
+       userfunc(
+               1, 2, 3, 4
+               );
+
+       userfunc(
+               1, 2,
+               3, 4
+               );
+
+       userfunc(
                1,
                2,
-               3,
-               4
+               3, 4
+               );
+
+       userfunc(
+               1,
+               2,
+               3, 4
+               );
+
+       userfunc
+       (
+               1, 2, 3, 4
        );
 
        userfunc
        (
                1,
                2,
-               3,
-               4
+               3, 4
        );
 
        userfunc
        (
                1,
                2,
-               3,
-               4
-       );
-
-       userfunc
-       (
-               1,
-               2,
-               3,
-               4
-       );
-
-       userfunc
-       (
-               1,
-               2,
-               3,
-               4
-       );
-
-       userfunc
-       (
-               1,
-               2,
-               3,
-               4
-       );
-
-       userfunc
-       (
-               1,
-               2,
-               3,
-               4
+               3, 4
        );
 }
MISMATCH: c:00730
diff --git a/home/fwojcik/progs/uncrustify/tests/expected/c/00731-nl_func_call.c b/home/fwojcik/progs/uncrustify/build/tests/results/c/00731-nl_func_call.c
index 38c17e45..b0296df0 100644
--- a/home/fwojcik/progs/uncrustify/tests/expected/c/00731-nl_func_call.c
+++ b/home/fwojcik/progs/uncrustify/build/tests/results/c/00731-nl_func_call.c
@@ -1,69 +1,55 @@
 void example(int a, int b, int c, int d);
 void example2(void);
-void userfunc
-(
-       int w,
-       int x,
-       int y,
-       int z
-);
+void userfunc(
+       int w, int x, int y, int z
+       );
 
 int main(void) {
-       example
-       (
+       example(
+               1, 2, 3, 4
+               );
+
+       example(
+               1, 2,
+               3, 4
+               );
+
+       example(
                1,
                2,
-               3,
-               4
+               3, 4
+               );
+
+       example(
+               1,
+               2,
+               3, 4
+               );
+
+       example
+       (
+               1, 2, 3, 4
        );
 
        example
        (
                1,
                2,
-               3,
-               4
+               3, 4
        );
 
        example
        (
                1,
                2,
-               3,
-               4
+               3, 4
        );
 
-       example
-       (
-               1,
-               2,
-               3,
-               4
-       );
+       example2(
+               );
 
-       example
-       (
-               1,
-               2,
-               3,
-               4
-       );
-
-       example
-       (
-               1,
-               2,
-               3,
-               4
-       );
-
-       example
-       (
-               1,
-               2,
-               3,
-               4
-       );
+       example2(
+               );
 
        example2
        (
@@ -73,67 +59,43 @@ int main(void) {
        (
        );
 
-       example2
-       (
-       );
+       userfunc(
+               1, 2, 3, 4
+               );
 
-       example2
+       userfunc(
+               1, 2,
+               3, 4
+               );
+
+       userfunc(
+               1,
+               2,
+               3, 4
+               );
+
+       userfunc(
+               1,
+               2,
+               3, 4
+               );
+
+       userfunc
        (
+               1, 2, 3, 4
        );
 
        userfunc
        (
                1,
                2,
-               3,
-               4
+               3, 4
        );
 
        userfunc
        (
                1,
                2,
-               3,
-               4
-       );
-
-       userfunc
-       (
-               1,
-               2,
-               3,
-               4
-       );
-
-       userfunc
-       (
-               1,
-               2,
-               3,
-               4
-       );
-
-       userfunc
-       (
-               1,
-               2,
-               3,
-               4
-       );
-
-       userfunc
-       (
-               1,
-               2,
-               3,
-               4
-       );
-
-       userfunc
-       (
-               1,
-               2,
-               3,
-               4
+               3, 4
        );
 }
MISMATCH: c:00731
diff --git a/home/fwojcik/progs/uncrustify/tests/expected/c/00732-nl_func_call.c b/home/fwojcik/progs/uncrustify/build/tests/results/c/00732-nl_func_call.c
index b57470e8..96d02a31 100644
--- a/home/fwojcik/progs/uncrustify/tests/expected/c/00732-nl_func_call.c
+++ b/home/fwojcik/progs/uncrustify/build/tests/results/c/00732-nl_func_call.c
@@ -7,15 +7,19 @@ int main(void) {
 
        example(1, 2,3, 4);
 
-       example(1,2,3, 4);
+       example(
+               1,2,3, 4);
 
-       example(1,2,3, 4);
+       example(
+               1,2,3, 4);
 
        example(1, 2, 3, 4);
 
-       example(   1,2,3, 4);
+       example(
+               1,2,3, 4);
 
-       example(   1,2,3, 4);
+       example(
+               1,2,3, 4);
 
        example2();
 
@@ -27,15 +31,27 @@ int main(void) {
 
        userfunc(1, 2, 3, 4);
 
-       userfunc(1, 2,3, 4);
+       userfunc(1, 2,
+                3, 4);
 
-       userfunc(1,2,3, 4);
+       userfunc(1,
+                2,
+                3, 4);
 
-       userfunc(1,2,3, 4);
+       userfunc(1,
+                2,
+                3, 4);
 
-       userfunc(1, 2, 3, 4);
+       userfunc
+               (1, 2, 3, 4);
 
-       userfunc(   1,2,3, 4);
+       userfunc
+               (   1,
+               2,
+               3, 4);
 
-       userfunc(   1,2,3, 4);
+       userfunc
+               (   1,
+               2,
+               3, 4);
 }
MISMATCH: c:00732
diff --git a/home/fwojcik/progs/uncrustify/tests/expected/c/00733-nl_func_call.c b/home/fwojcik/progs/uncrustify/build/tests/results/c/00733-nl_func_call.c
index 2290132d..db35e838 100644
--- a/home/fwojcik/progs/uncrustify/tests/expected/c/00733-nl_func_call.c
+++ b/home/fwojcik/progs/uncrustify/build/tests/results/c/00733-nl_func_call.c
@@ -16,23 +16,28 @@ int main(void) {
                2,
                3, 4);
 
-       example(1, 2, 3, 4);
+       example
+               (1, 2, 3, 4);
 
-       example(   1,
-                  2,
-                  3, 4);
+       example
+               (   1,
+               2,
+               3, 4);
 
-       example(   1,
-                  2,
-                  3, 4);
+       example
+               (   1,
+               2,
+               3, 4);
 
        example2();
 
        example2();
 
-       example2();
+       example2
+               ();
 
-       example2(    );
+       example2
+               (    );
 
        userfunc(1, 2, 3, 4);
 
@@ -47,13 +52,16 @@ int main(void) {
                 2,
                 3, 4);
 
-       userfunc(1, 2, 3, 4);
+       userfunc
+               (1, 2, 3, 4);
 
-       userfunc(   1,
-                   2,
-                   3, 4);
+       userfunc
+               (   1,
+               2,
+               3, 4);
 
-       userfunc(   1,
-                   2,
-                   3, 4);
+       userfunc
+               (   1,
+               2,
+               3, 4);
 }
MISMATCH: c:00733
diff --git a/home/fwojcik/progs/uncrustify/tests/expected/c/00734-nl_func_call.c b/home/fwojcik/progs/uncrustify/build/tests/results/c/00734-nl_func_call.c
index 8233b0ec..8b0fa3bb 100644
--- a/home/fwojcik/progs/uncrustify/tests/expected/c/00734-nl_func_call.c
+++ b/home/fwojcik/progs/uncrustify/build/tests/results/c/00734-nl_func_call.c
@@ -1,9 +1,6 @@
 void example(int a, int b, int c, int d);
 void example2(void);
-void userfunc(int w,
-              int x,
-              int y,
-              int z);
+void userfunc(int w, int x, int y, int z);
 
 int main(void) {
        example(1,
@@ -62,47 +59,35 @@ int main(void) {
        (
        );
 
-       userfunc(1,
-                2,
-                3,
-                4);
+       userfunc(1, 2, 3, 4);
 
-       userfunc(1,
-                2,
-                3,
-                4);
+       userfunc(1, 2,
+                3, 4);
 
        userfunc(
                1,
                2,
-               3,
-               4);
+               3, 4);
 
        userfunc(
                1,
                2,
-               3,
-               4
+               3, 4
                );
 
        userfunc
-               (1,
-               2,
-               3,
-               4);
+               (1, 2, 3, 4);
 
        userfunc
        (
                1,
                2,
-               3,
-               4);
+               3, 4);
 
        userfunc
        (
                1,
                2,
-               3,
-               4
+               3, 4
        );
 }
MISMATCH: c:00734
393 / 398 tests passed
5 tests did not match the expected outputm
```

### CODE REVIEW

**Review by micheleCTDEAdmin — APPROVED** [review]
Excellent work, I can see lot of improvement in my big test repo!
Well done!!

**Comment by micheleCTDEAdmin** [comment]
@fwojcik 
btw, if you want to have access to my 3200 test repo for additional tests, let me know.

**Comment by fwojcik** [comment]
Thank you! I'm glad to be helping. And, yes, I would like access to your additional tests. I'm about 75% done with the biggest non-enhancement change that I currently plan to submit, and an additional regression testing resource would be great.

**Comment by fwojcik** [comment]
Another huge help for me right now would be to fix issue #3783. It prevents me from ever getting a truly clean ctest run.

**Comment by micheleCTDE** [comment]
You can find my test repo [here](https://mirror.git.trinitydesktop.org/gitea/MicheleC/styleTest)

master branch is the clean starting point, then I usually make a branch and do 3 runs of uncrustify on all the files with the config stored in the file .uncrustify_tde.cfg (included in each test branch).
Branch v40-dc69f673 is the latest one and done with the code of this PR included.

Feel free to clone the repo and do whatever testing you want and with whatever config :-)
